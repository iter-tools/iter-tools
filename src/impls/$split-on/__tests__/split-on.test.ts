/**
 * @generated-from ./$split-on.test.ts
 * This file is autogenerated from a template. Please do not edit it directly.
 * To rebuild it from its template use the command
 * > npm run generate
 * More information can be found in CONTRIBUTING.md
 */

import { splitOn } from 'iter-tools-es';
import { wrap, unwrapDeep } from '../../../test/helpers.js';

describe('splitOn', () => {
  it('should split between every value which is equal to the on argument', () => {
    expect(unwrapDeep(splitOn(null, wrap([1, null, 2, null, 3])))).toEqual([[1], [2], [3]]);
  });

  it('should throw when splits are consumed out of order', () => {
    const parts = splitOn(null, wrap([1, null, 2]));
    const a = parts.next().value;
    const b = parts.next().value;

    expect(
      (() => {
        try {
          unwrapDeep([b, a]);
        } catch (e) {
          return e;
        }
      })(),
    ).toMatchSnapshot();
  });

  it('should yield [] between two separators', () => {
    expect(unwrapDeep(splitOn(null, wrap([1, null, null, 3])))).toEqual([[1], [], [3]]);
  });

  it('should yield [], [] when only separator', () => {
    expect(unwrapDeep(splitOn(null, wrap([null])))).toEqual([[], []]);
  });

  it('passes through the empty iterable', () => {
    expect(unwrapDeep(splitOn(0, null))).toEqual([]);
  });

  describe('when same function is specified', () => {
    const same = (a: number, b: number) => Math.abs(a) === Math.abs(b);
    it('uses same value to do comparison', () => {
      expect(unwrapDeep(splitOn(same, 2, wrap([1, 2, 3])))).toEqual([[1], [3]]);
      expect(unwrapDeep(splitOn(() => false, 2, wrap([1, 2, 3])))).toEqual([[1, 2, 3]]);
    });
  });

  describe('when source is a string', () => {
    it('warns', () => {
      splitOn(null, 'abc');
      expect(console.warn).callsMatchSnapshot();
    });
  });
});
