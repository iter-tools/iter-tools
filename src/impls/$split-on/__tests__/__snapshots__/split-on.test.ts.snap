// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`splitOn should throw when splits are consumed out of order 1`] = `[Error: Cannot take from this split part. It is no longer the active part.]`;

exports[`splitOn when source is a string warns 1`] = `
Array [
  Array [
    "For string inputs use splitOnSeq instead of splitOn",
  ],
]
`;
